src/integrators/sdirk_integrators.jl

```diff
@@ -961,155 +997,158 @@ end
     end
   end
 
+  # FSAL
   @. zprev = dt*integrator.fsalfirst
 
+  ##### Solve Trapezoid Step
+
   # TODO: Add extrapolation
   @. zᵧ = zprev
-  iter = 1
 
-  @. uᵧ = (uprev + d*zprev) + d*zᵧ
-  f(t+γdt,uᵧ,k)
-  @. k = dt*k - zᵧ
+  # initial step of Newton iteration
+  iter = 1
+  tstep = t + γ*dt
+  @. tmp = uprev + d*zprev
+  @. u = tmp + d*zᵧ
+  f(tstep,u,k)
+  @. b = dt*k - zᵧ
   if has_invW(f)
-    A_mul_B!(vec(Δzᵧ),W,vec(k)) # Here W is actually invW
+    A_mul_B!(vec(dz),W,vec(b)) # Here W is actually invW
   else
-    integrator.alg.linsolve(vec(Δzᵧ),W,vec(k),new_W)
+    integrator.alg.linsolve(vec(dz),W,vec(b),new_W)
   end
-  ndz = integrator.opts.internalnorm(Δzᵧ)
-  @. zᵧ = zᵧ + Δzᵧ
-
-  @. uᵧ = (uprev + d*zprev) + d*zᵧ
+  ndz = integrator.opts.internalnorm(dz)
+  zᵧ .+= dz
 
   η = max(cache.ηold,eps(first(u)))^(0.8)
-  if integrator.success_iter > 0
-    do_newton = (η*ndz > κ*tol)
-  else
-    do_newton = true
-  end
+  do_newton = integrator.success_iter == 0 || η*ndz > κtol
 
+  # Newton iteration
   fail_convergence = false
   while (do_newton || iter < integrator.alg.min_newton_iter) && iter < integrator.alg.max_newton_iter
     iter += 1
-    @. uᵧ = (uprev + d*zprev) + d*zᵧ
-    f(t+γdt,uᵧ,k)
-    @. k = dt*k - zᵧ
+    @. u = tmp + d*zᵧ
+    f(tstep,u,k)
+    @. b = dt*k - zᵧ
     if has_invW(f)
-      A_mul_B!(vec(Δzᵧ),W,vec(k)) # Here W is actually invW
+      A_mul_B!(vec(dz),W,vec(b)) # Here W is actually invW
     else
-      integrator.alg.linsolve(vec(Δzᵧ),W,vec(k),false)
+      integrator.alg.linsolve(vec(dz),W,vec(b),false)
     end
     ndzprev = ndz
-    ndz = integrator.opts.internalnorm(Δzᵧ)
+    ndz = integrator.opts.internalnorm(dz)
     θ = ndz/ndzprev
-    if θ > 1 || ndz*(θ^(integrator.alg.max_newton_iter - iter)/(1-θ)) > κ*tol
+    if θ > 1 || ndz*(θ^(integrator.alg.max_newton_iter - iter)/(1-θ)) > κtol
       fail_convergence = true
       break
     end
     η = θ/(1-θ)
-    do_newton = (η*ndz > κ*tol)
-    @. zᵧ = zᵧ + Δzᵧ
+    do_newton = (η*ndz > κtol)
+    zᵧ .+= dz
   end
 
   if (iter >= integrator.alg.max_newton_iter && do_newton) || fail_convergence
     integrator.force_stepfail = true
     return
   end
 
-  @. uᵧ = (uprev + d*zprev) + d*zᵧ
-
   ################################## Solve BDF2 Step
 
   ### Initial Guess From Shampine
-  a1 = (1.5 + sqrt(2)); a2 = (2.5 + 2sqrt(2)); a3 = (6 + 4.5sqrt(2))
-  @. z = a1*zprev + a2*zᵧ - a3*(uᵧ - uprev)
-  iter = 1
+  z = @. α1*zprev + α2*zᵧ
```

Comment:
Yes, I documented it: https://github.com/devmotion/OrdinaryDiffEq.jl/blob/a0150ee09d8581302ad89343d9750f0e59965693/src/tableaus/sdirk_tableaus.jl#L26-L33